File: ContainerWithMostWater.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int maxVolume = 0;
        while (l < r) {
            maxVolume = max(maxVolume, (r-l)*min(height[l],height[r]));
            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return maxVolume;
    }
};

int main() {}

File: Blind75/Arrays/ContainerWithMostWater.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        result = 0
        l, r = 0, len(height) - 1

        while l < r:
            result = max(result, (r - l) * min(height[l], height[r]))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return result
